internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go

```diff
@@ -0,0 +1,631 @@
+			}
+
+			if metadata.ResourceData.HasChange("coordinator_vcore_count") {
+				parameters.Properties.CoordinatorVCores = &model.CoordinatorVCoreCount
+			}
+
+			if metadata.ResourceData.HasChange("ha_enabled") {
+				parameters.Properties.EnableHa = &model.HaEnabled
+			}
+
+			if metadata.ResourceData.HasChange("maintenance_window") {
+				parameters.Properties.MaintenanceWindow = expandMaintenanceWindow(model.MaintenanceWindow)
+			}
+
+			if metadata.ResourceData.HasChange("node_count") {
+				parameters.Properties.NodeCount = &model.NodeCount
+			}
+
+			if metadata.ResourceData.HasChange("node_public_ip_access_enabled") {
+				parameters.Properties.NodeEnablePublicIPAccess = &model.NodePublicIPAccessEnabled
+			}
+
+			if metadata.ResourceData.HasChange("node_server_edition") {
+				parameters.Properties.NodeServerEdition = &model.NodeServerEdition
+			}
+
+			if metadata.ResourceData.HasChange("node_storage_quota_in_mb") {
+				parameters.Properties.NodeStorageQuotaInMb = utils.Int64(model.NodeStorageQuotaInMb)
+			}
+
+			if metadata.ResourceData.HasChange("node_vcores") {
+				parameters.Properties.NodeVCores = utils.Int64(model.NodeVCores)
+			}
+
+			if metadata.ResourceData.HasChange("preferred_primary_zone") {
+				parameters.Properties.PreferredPrimaryZone = &model.PreferredPrimaryZone
+			}
+
+			if metadata.ResourceData.HasChange("shards_on_coordinator_enabled") {
+				parameters.Properties.EnableShardsOnCoordinator = &model.ShardsOnCoordinatorEnabled
+			}
+
+			if metadata.ResourceData.HasChange("sql_version") {
+				parameters.Properties.PostgresqlVersion = &model.SqlVersion
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				parameters.Tags = &model.Tags
+			}
+
+			if err := client.UpdateThenPoll(ctx, *id, parameters); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CosmosDbPostgreSQLClusterResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cosmos.ClustersClient
+
+			id, err := clusters.ParseServerGroupsv2ID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			state := CosmosDbPostgreSQLClusterModel{
+				Name:              id.ServerGroupsv2Name,
+				ResourceGroupName: id.ResourceGroupName,
+				Location:          location.Normalize(model.Location),
+			}
+
+			if props := model.Properties; props != nil {
+				state.AdministratorLoginPassword = metadata.ResourceData.Get("administrator_login_password").(string)
```

Comment:
TF would fail with below error after it's removed. I remember the colleague from your team previously suggested me to read it from tf file.

```
=== RUN   TestAccCosmosDbPostgreSQLCluster_basic
=== PAUSE TestAccCosmosDbPostgreSQLCluster_basic
=== CONT  TestAccCosmosDbPostgreSQLCluster_basic
    testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
                
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_cosmosdb_postgresql_cluster.test will be updated in-place
          ~ resource "azurerm_cosmosdb_postgresql_cluster" "test" {
              + administrator_login_password         = (sensitive value)
                id                                   = "/subscriptions/xx-xx-xxx-xx/resourceGroups/acctestRG-pshsc-230426160316468761/providers/Microsoft.DBforPostgreSQL/serverGroupsv2/acctestcluster230426160316468761"
                name                                 = "acctestcluster230426160316468761"
                # (15 unchanged attributes hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
```

Reply:
Oh. That's interesting. I haven't seen that before. I wonder whats going on there. But no worries, we can keep it that way!